pkg/api/controller.go

```diff
@@ -802,27 +810,76 @@ func (c *Controller) ListUsers(w http.ResponseWriter, r *http.Request, params Li
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) generateResetPasswordToken(email string) (*string, error) {
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
+	if err != nil {
+		return nil, err
+	}
+	return &token, nil
+}
+
+func (c *Controller) inviteUserRequest(email string) error {
+	// TODO (@shimi9276) create template for sending the email with link for invite using the passed template
+	token, err := c.generateResetPasswordToken(email)
+	if err != nil {
+		return err
+	}
+	err = c.Emailer.SendEmailWithLimit([]string{email}, *token, *token, nil)
+	if err != nil {
+		return err
+	} else {
+		c.Logger.WithField("email", email).Info("invite email sent")
+	}
+	return nil
+}
+
 func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body CreateUserJSONRequestBody) {
+	invite := swag.BoolValue(body.InviteUser)
+	id := body.Id
+	var addr *mail.Address
+	var err error
+	if invite {
+		addr, err = mail.ParseAddress(body.Id)
+		if handleAPIError(w, err) {
```

Comment:
handleAPIError will not know how to handle invalid address error - in case of any error, you need to write back validation error.
The error we write here should describe to the caller that the invite user id is in invalid email format.

Reply:
I did check this one out and tried to invite a user with an empty email, it returned 500 (internal server error) with the proper error (i.e. no address), so I think it should be pretty clear what the error is, In any case, I replaced it with a more descriptive response as you suggested.